src/rust/uk.mangainua/src/lib.rs

```diff
@@ -0,0 +1,429 @@
+			Ok(n) => total = n,
+			_ => total = 0,
+		}
+	}
+
+	Ok(MangaPageResult {
+		manga: manga_arr,
+		has_more: page < total,
+	})
+}
+
+#[get_manga_details]
+fn get_manga_details(id: String) -> Result<Manga> {
+	let html = Request::new(id.as_str(), HttpMethod::Get)
+		.html()
+		.expect("get manga details html array not an array of nodes");
+
+	let title = html.select(".UAname").text().read();
+	let cover = html
+		.select(".item__full-sidebar--poster img")
+		.attr("abs:src")
+		.read();
+	let description = html.select(".item__full-description").text().read();
+
+	let mut is_nsfw = false;
+	let mut status = "Unknown";
+
+	let mut categories = Vec::new();
+	for categ in html.select(".item__full-sidebar--description a").array() {
+		let categ_node = categ.as_node().expect("");
+		let name = categ_node.text().read();
+
+		if categories.contains(&name) {
+			// categories are duplicated on the website
+			continue;
+		}
+		categories.push(name.clone());
+
+		if !is_nsfw {
+			// check if only didnt find nsfw category
+			is_nsfw = helper::is_nsfw(name.clone());
+		}
+		if status == "Unknown" {
+			// check if only didnt find status
+			status = helper::get_status_string(name.clone());
+		}
+	}
+
+	let status_res: MangaStatus = match status {
+		"Ongoing" => MangaStatus::Ongoing,
+		"Unknown" => MangaStatus::Unknown,
+		"Completed" => MangaStatus::Completed,
+		"Покинуто" => MangaStatus::Cancelled,
+		_ => MangaStatus::Unknown,
+	};
+
+	let manga = Manga {
+		id,
+		cover,
+		title,
+		description,
+		categories: categories.clone(),
+		status: status_res,
+		nsfw: if is_nsfw {
+			MangaContentRating::Nsfw
+		} else {
+			MangaContentRating::Safe
+		},
+		viewer: MangaViewer::Rtl,
+		..Default::default()
+	};
+	Ok(manga)
+}
+
+#[get_chapter_list]
+fn get_chapter_list(id: String) -> Result<Vec<Chapter>> {
+	let html = Request::new(id.as_str(), HttpMethod::Get)
+		.html()
+		.expect("chapter html array not an array of nodes");
+
+	let mut chapter: f32 = -1.0;
+
+	let mut res: Vec<Chapter> = Vec::new();
+	for result in html.select(".linkstocomicsblockhidden .ltcitems").array() {
+		let res_node = result.as_node().expect("");
+		let href = res_node.select("a").attr("href").read();
+		let mut title = res_node.select("a").text().read();
+
+		let date = res_node
+			.select(".ltcright")
+			.first()
+			.text()
+			.0
+			.as_date("dd.MM.yyyy", None, None)
+			.unwrap_or(-1.0);
+
+		let scanlator = res_node
+			.select(".ltcright .tooltip .tooltiptext")
+			.text()
+			.read();
+
+		// Alternative translation don't have chapter number. use previous parsed
+		if title.contains("Альтернативний переклад") {
+			title = String::from("↑") + &title;
+		} else {
+			// parse volume and chapter.
+			let volume_chapter = title.clone();
+			let replaced = volume_chapter.replace("НОВЕ ", "");
+			let arr: Vec<_> = replaced.split_whitespace().collect();
+
+			match arr[3].parse::<f32>() {
+				Ok(n) => chapter = n,
+				_e => continue,
```

Comment:
maybe consider falling back to -1? that way, if it fails to find a chapter number, it should still return the chapter with no number rather than just skipping it. I'm not sure how the site works though
```suggestion
				_ => continue,
```